#19 Updated XIVStats-Gatherer-Java for Stormblood

matthewhillier opened the pull request:
Updated the XIVStats program to support the Stormblood updates to the Lodestone.

- Added Omega & Louisiox servers
- Added Scholar
- Added Red Mage
- Added Samurai
- Added Stormblood Pre-Order Check
- Added Stormblood Completion Check
- - Note: This is a bit of a guess because the final MSQ doesn't drop a mount. Instead, looking for drops from Kugane Castle, Temple of the Fist or Deltascape V4 as the only real way to determine passing beyond the final quest

matthewhillier committed:
#19 Adding in variables for Scholar, Red Mage and Samurai and updating the parse order to match the Lodestone update

matthewhillier committed:
#19 Added Omega & Louisoix servers

matthewhillier committed:
#19 Updated Player Unit Test

matthewhillier committed:
#19 Added in completion stats for Stormblood & Corrected parameter ordering in the getPlayerTest

NOTE: SB completion is currently opaque as the MSQ final quest doesn't drop anything. Implemeted a best-guess based on end-game content drops.

matthewhillier committed:
#19 Added Table fields and mappings for Stormblood progression

matthewhillier committed:
#19 Added level fields for Scholar, Red Mage and Samurai

matthewhillier committed:
#19 Added missing carpenter field

matthewhillier committed:
#19 Corrected missing comma

matthewhillier committed:
Update README.md

Added changes to the DB structure

matthewhillier committed:
Update README.md

Corrected spelling of "Ivon"

ReidWeb commented:
Hey,

Thanks so much for submitting this! I'll be taking a look at the weekend, I need to adjust the CI a bit

matthewhillier commented:
No worries - I had fun getting eclipse to run these tests consistently. Might want to consider using an H2 in-memory DB for running the Unit Tests here - testing against a real DB is likely to give you issues if this ever gets more complex.

matthewhillier committed:
Adding ignores for Eclipse settings files.

ReidWeb commented on src/main/java/com/ffxivcensus/gatherer/GathererController.java:
```diff
@@ -578,6 +581,7 @@ protected String writeToDB(Player player) {
             sbValues.append(");");
 
             String strSQL = sbFields.toString() + sbValues.toString();
+            System.out.println(strSQL);
```
Please remove this print statement

matthewhillier replied:
Done

ReidWeb commented on src/test/java/com/ffxivcensus/gatherer/PlayerTest.java:
```diff
@@ -72,72 +79,78 @@ public void testGetPlayer() throws Exception {
         assertTrue(playerOne.isHasCompletedHW());
-        assertEquals(playerOne.getBitHasCompletedHW(), 1);
+        assertEquals(1, playerOne.getBitHasCompletedHW());
+        // Currently no way to definitively tell player has completed the SB Main Scenario
```
Myself and @Pricetx are not fond of the 'hit and miss nature' of this way of evaluating 4.0 MSQ completion, please remove this logic

ReidWeb reviewed (commented):
Tests are failing in CI, but locally different tests are failing....I'm not entirely sure why and can't really investigate at this time.

matthewhillier committed:
Fixed some code review feedback:
- Removed the SB completion check
- Removed extraneous SQL logging

matthewhillier committed:
Removed unnecessary stream hijacking.

matthewhillier commented:
Fixed the immediate code review issues.

The test results are... inconsistent :)

- Running `maven test` locally completes fine with no issues
- Running them in Eclipse is fine up until ConsoleTest (they're running reverse order from maven)

The issue with Eclipse is straight-forward - `GathererControllerTest` (and `ConsoleTest`) both override the `System.out` and `System.err` buffers at the start of a test run and then set it to null at the end. This causes later tests to fail in Eclipse as the test-runner here shares the same JVM, thus later tests are being passed a "_broken_" JVM (where `System.out` and `System.err` are `null`).

`GathererControllerTest` doesn't actually appear to use these buffers, so I've removed them from this test class to reduce the risk of running into this as a false-positive for anyone else.

The issue in the CI appears to be an issue with the `config.xml` file on the CI server. I can't see what's wrong, per-se, as the Jenkins script seems to suggest the file is being copied from somewhere locally on the server, however the message being thrown - `[Fatal Error] config.xml:1:7: The processing instruction target matching "[xX][mM][lL]" is not allowed.` - would suggest an issue with the format of the XML file, specifically relating to the `<?xml ?>` declaration in the file. Stackoverflow points to a couple of possible reasons for this: [here](https://stackoverflow.com/questions/19889132/error-the-processing-instruction-target-matching-xxmmll-is-not-allowed).

matthewhillier committed:
Removed the Stormblood warnining from the README.md file.

ReidWeb reviewed (approved):
Approving for merge
